src/components/Observations/map/ObservationMapView.tsx

```diff
@@ -48,13 +56,31 @@ export default function ObservationMapView({ observationsResults }: ObservationM
     }
 
     return MapService.getMapDataFromObservation(selectedObservation);
-  }, [selectedObservationDate, observationsResults, plantingSiteMapData]);
+  }, [selectedObservation, selectedObservationDate, plantingSiteMapData]);
   useEffect(() => {
     if (!plantingSiteMapData && mapData.site) {
       setPlantingSiteMapData(mapData.site);
     }
   }, [mapData, plantingSiteMapData]);
 
+  const defaultTimeZone = useDefaultTimeZone();
+  const details = useAppSelector((state) =>
```

Comment:
Is this app selector mostly to re-apply the search to the planting zones?
Thoughts about doing this at the parent selector level (maybe a boolean option to the selector - set to true only for mapview)? Basically [here](https://github.com/terraware/terraware-web/blob/main/src/redux/features/observations/observationsSelectors.ts#L42), pass the boolean option further down into searchZones?
Will simplify things a bit and will help more when I add search filters.
Otherwise might be good to just separate the search out inside list and map views separately.

Reply:
Ok, I think the search filters will work here the same way search works. Should be just fine. Ignore my previous suggestion.

Reply:
The selector was to get the search-filtered set of planting zones. I don't dispatch the thunk here, so I don't think it's causing any additional work, just reading the data from state. But I do see what you mean, and maybe should have created a new selector. Thanks for the suggestions here. I can leave this in place for now, but may replace it in subsequent iterations.

Reply:
Been thinking more about this. Figured I'd explain why my original suggestion and then let you make a call.
The map view is a full-depth view of the planting site observations. 
The observation details object passed into the map component has full depth information, figured that might be sufficient to render the map. 
Any search requirements could potentially be applied to the full depth content of the observations, and removes the need to select further data.
Hence my original suggestion. 
It is not a functional thing, more from a data provider/architecture point of view. 
It is not the only way, what we have here is just as good.

Reply:
Ohh, gotcha, I understand. So `observationResults` contains all the zones for the matched observations, and I don't want to exclude zones from the map if they don't match search. But i do need to figure out which zones were matched to apply highlights and focus, which is why I added this additional selector. However, since the `searchObservationDetails` selector contains more information than is needed (much of it duplicated in `observationResults`), it might have been better to just select the list of search-matched zones here.

Reply:
Between selecting versus doing the search within the component - delegating to the selector (keep rendering separate) like you have in this PR seems better for separation (use redux for data views).
I was thinking along the lines of doing a hierarchical search within the selector code (if the map view is enabled) - but that requires re-selecting when user switches back and forth between list and map views -  seems wasteful, didn't think about that.
Looks like what you have here is probably best middle ground, we could revisit once all the search filters etc are implemented (next on my plate)?